spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplateTests.java

```diff
@@ -365,6 +373,46 @@ public void testNestedTxBinding() throws Exception {
 		verify(channel1).txCommit();
 	}
 
+	@Test
+	public void testShutdownWhileWaitingForReply() throws Exception {
+		ConnectionFactory mockConnectionFactory = mock(ConnectionFactory.class);
+		Connection mockConnection = mock(Connection.class);
+		Channel mockChannel = mock(Channel.class);
+
+		given(mockConnectionFactory.newConnection(any(ExecutorService.class), anyString())).willReturn(mockConnection);
+		given(mockConnection.isOpen()).willReturn(true);
+		given(mockConnection.createChannel()).willReturn(mockChannel);
+		given(mockChannel.queueDeclare()).willReturn(new AMQImpl.Queue.DeclareOk("foo", 0, 0));
+		final AtomicReference<ShutdownListener> listener = new AtomicReference<>();
+		final CountDownLatch shutdownLatch = new CountDownLatch(1);
+		willAnswer(invocation -> {
+			listener.set(invocation.getArgument(0));
+			shutdownLatch.countDown();
+			return null;
+		}).given(mockChannel).addShutdownListener(any());
+		SingleConnectionFactory connectionFactory = new SingleConnectionFactory(mockConnectionFactory);
+		connectionFactory.setExecutor(mock(ExecutorService.class));
+		RabbitTemplate template = new RabbitTemplate(connectionFactory);
+		template.setReplyTimeout(60_000);
+		Message input = new Message("Hello, world!".getBytes(), new MessageProperties());
+		Executors.newSingleThreadExecutor().execute(() -> {
```

Comment:
Don't we need to kill the thread pool in the end of test?
The point is because of this:

```
public static ExecutorService newSingleThreadExecutor() {
        return new FinalizableDelegatedExecutorService
            (new ThreadPoolExecutor(1, 1,
                                    0L, TimeUnit.MILLISECONDS,
                                    new LinkedBlockingQueue<Runnable>()));
    }
```
I mean it dies, of course, in the end of the JVM, but it still going to hang on during test suit execution.